### PR TITLE
Status check

### PR DESCRIPTION
**Pull request by GVardakis**
With this pull request, flask (app.py) can access the state variables showing at each time what processes are active. This is done through environment variables. The variables provided are:
- BACKEND_TX_PID : Pid of process listening to UI for transmissions to the backend
- BACKEND_RX_PID : Pid of process listening for frames from backend
- BACKEND : Current backend used ('gnuradio' or 'serial')
- MODE : Current mode used ('network' or 'cmd_ctrl' )
- ECSS_FEEDER_PID : Process waiting for requests from UI for providing the frames that arrived from backend
- TASK_FEEDER_PID : Process waiting for requests from UI for providing upcoming tasks for display
- SCHEDULER : 'ON' if scheduler is running, 'OFF' otherwise

The default MODE at the start of execution is 'network'

### CODE REVIEW

**Comment by ppapadeas** [comment]
please rebase to fetch the pep8 changes committed on dev branch last night. Also make sure to follow flake8 moving forward (it is as simple as "flake8 ." on your source dir)

**Comment by ppapadeas** [comment]
Trying to run the branch got me:
Registering `get_jobs` periodic task (5 min. interval)
Registering `post_data` periodic task (15 min. interval)
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib64/python2.7/threading.py", line 804, in **bootstrap_inner
    self.run()
  File "/usr/lib64/python2.7/threading.py", line 757, in run
    self.__target(_self.__args, *_self.__kwargs)
  File "/home/liknus/git/satnogs/satnogs-client/satnogsclient/scheduler/tasks.py", line 259, in status_listener
    os.environ['TASK_FEEDER_PID'] = tf.pid
  File "/home/liknus/git/envs/satnogs-client/lib64/python2.7/os.py", line 473, in __setitem**
    putenv(key, item)
TypeError: must be string, not int

**Comment by ppapadeas** [comment]
Please rebase to dev

File: satnogsclient/scheduler/tasks.py [hunk 1]
```diff
@@ -19,15 +19,21 @@
...
 from satnogsclient.upsat import serial_handler
 from satnogsclient.upsat.gnuradio_handler import write_to_gnuradio, read_from_gnuradio
 from time import sleep
 
-
 logger = logging.getLogger('satnogsclient')
 
 
+def signal_term_handler(signal, frame):
+    dict = frame.f_locals
```
Comment (johngian) [inline satnogsclient/scheduler/tasks.py]: `dict` is a built-in type in python. Here you are overriding the dictionary definition with something else.

File: satnogsclient/scheduler/tasks.py [hunk 2]
```diff
@@ -164,24 +170,31 @@ def get_jobs():
...
-            conn.recv(sock.tasks_buffer_size)
-            if not q.empty():
-                conn.send(q.get())
-            else:
-                conn.send('[]')
-    p.join()
+    child_pid = 0
+    try:
+        logger.info('Started task feeder')
+        print port1, ' ', port2
```
Comment (johngian) [inline satnogsclient/scheduler/tasks.py]: Please be consistent between using `logging` and `print`. You should use python logging infrastructure everywhere.

File: satnogsclient/scheduler/tasks.py [hunk 3]
```diff
@@ -164,24 +170,31 @@ def get_jobs():
...
+        while 1:
+            conn = sock.accept()
+            if conn:
+                conn.recv(sock.tasks_buffer_size)
+                if not q.empty():
+                    conn.send(q.get())
+                else:
+                    conn.send('[]')
+        p.join()
+    except:
```
Comment (johngian) [inline satnogsclient/scheduler/tasks.py]: Here you are hiding everything under this catch-all except. This means that if something but happens in `try` you will never know what went wrong.

Comment (johngian) [inline satnogsclient/scheduler/tasks.py]: Either be more specific in the exception you are catching here or use some other kind of logic to handle this.

File: satnogsclient/scheduler/tasks.py [hunk 4]
```diff
@@ -203,21 +216,27 @@ def task_listener(port, queue):
...
-    pr.start()
-    while 1:
-        conn = sock.recv()
-        list = []
-        while not qu.empty():
-            a = qu.get()
-            list.append(a)
-        sock.sendto(json.dumps(list), conn[1])
-    pr.join()
+    child_pid = 0
```
Comment (johngian) [inline satnogsclient/scheduler/tasks.py]: Exactly the same comments apply here.

File: satnogsclient/scheduler/tasks.py [hunk 5]
```diff
@@ -203,21 +216,27 @@ def task_listener(port, queue):
...
+    try:
```
Comment (johngian) [inline satnogsclient/scheduler/tasks.py:218]: Exactly the same comments apply here.

File: satnogsclient/scheduler/tasks.py [hunk 6]
```diff
@@ -234,84 +253,99 @@ def ecss_listener(port, queue):
 
 def status_listener():
     logger.info('Started upsat status listener')
+    logger.info('Starting scheduler...')
+    scheduler.start()
+    interval = settings.NETWORK_API_QUERY_INTERVAL
+    scheduler.add_job(get_jobs, 'interval', minutes=interval)
+    msg = 'Registering `get_jobs` periodic task ({0} min. interval)'.format(interval)
+    print msg
```
Comment (johngian) [inline satnogsclient/scheduler/tasks.py]: Please be consistent when logging.

File: satnogsclient/scheduler/tasks.py [hunk 7]
```diff
@@ -234,84 +253,99 @@ def ecss_listener(port, queue):
...
+    interval = settings.NETWORK_API_POST_INTERVAL
+    msg = 'Registering `post_data` periodic task ({0} min. interval)'.format(interval)
```
Comment (johngian) [inline satnogsclient/scheduler/tasks.py:261]: Same here.

File: satnogsclient/scheduler/tasks.py [hunk 8]
```diff
@@ -234,84 +253,99 @@ def ecss_listener(port, queue):
...
+    print msg
+    scheduler.add_job(post_data, 'interval', minutes=interval)
+    print 'Scheduler jobs are ', scheduler.print_jobs()
```
Comment (johngian) [inline satnogsclient/scheduler/tasks.py]: Same here.

File: satnogsclient/scheduler/tasks.py [hunk 9]
```diff
@@ -234,84 +253,99 @@ def ecss_listener(port, queue):
...
+    tf = Process(target=task_feeder, args=(settings.TASK_FEEDER_TCP_PORT, settings.TASK_LISTENER_TCP_PORT,))
+    tf.start()
+    os.environ['TASK_FEEDER_PID'] = str(tf.pid)
     sock = Udpsocket(('127.0.0.1', settings.STATUS_LISTENER_PORT))
+    os.environ['BACKEND_TX_PID'] = '0'
+    os.environ['BACKEND_RX_PID'] = '0'
+    os.environ['BACKEND'] = ""
```
Comment (johngian) [inline satnogsclient/scheduler/tasks.py:270]: nit: I suggest you always use single-quotes except in cases you really need double-quotes.

File: satnogsclient/scheduler/tasks.py [hunk 10]
```diff
@@ -234,84 +253,99 @@ def ecss_listener(port, queue):
...
+    os.environ['MODE'] = "network"
```
Comment (johngian) [inline satnogsclient/scheduler/tasks.py:271]: Same here.

File: satnogsclient/scheduler/tasks.py [hunk 11]
```diff
@@ -234,84 +253,99 @@ def ecss_listener(port, queue):
...
+    os.environ['ECSS_FEEDER_PID'] = '0'
+    os.environ['SCHEDULER'] = 'ON'
     while 1:
-        print 'Listening status, ', settings.STATUS_LISTENER_PORT
         conn = sock.recv()
         dict = json.loads(conn[0])
```
Comment (johngian) [inline satnogsclient/scheduler/tasks.py]: Same here. `dict` is built-in in python. You are overriding the dictionary definition.

File: satnogsclient/scheduler/tasks.py [hunk 12]
```diff
@@ -234,84 +253,99 @@ def ecss_listener(port, queue):
...
             if dict['backend'] == 'gnuradio':
-                status.BACKEND = 'gnuradio'
+                os.environ['BACKEND'] = 'gnuradio'
                 tx = Process(target=write_to_gnuradio, args=())
-                print 'Started process'
                 tx.daemon = True
                 tx.start()
-                print status.BACKEND_TX_PID
-                status.BACKEND_TX_PID = tx.pid
+                print 'Started gnuradio tx process ', tx.pid
```
Comment (johngian) [inline satnogsclient/scheduler/tasks.py]: Same here.

File: satnogsclient/scheduler/tasks.py [hunk 13]
```diff
@@ -234,84 +253,99 @@ def ecss_listener(port, queue):
...
+                os.environ['BACKEND_TX_PID'] = str(tx.pid)
                 rx = Process(target=read_from_gnuradio, args=())
                 rx.daemon = True
                 rx.start()
-                status.BACKEND_RX_PID = rx.pid
+                print 'Started gnuradio rx process ', rx.pid
```
Comment (johngian) [inline satnogsclient/scheduler/tasks.py]: Same here.

File: satnogsclient/scheduler/tasks.py [hunk 14]
```diff
@@ -234,84 +253,99 @@ def ecss_listener(port, queue):
...
                 rx.start()
-                status.BACKEND_RX_PID = rx.pid
-        if 'switch_mode' in dict.keys() and dict['switch_mode']:
-            print 'Changing mode'
-            if dict['mode'] == status.MODE:
+                os.environ['BACKEND_RX_PID'] = str(rx.pid)
+        if 'mode' in dict.keys():
+            if dict['mode'] == os.environ['MODE']:
                 continue
+            print 'Changing mode'
```
Comment (johngian) [inline satnogsclient/scheduler/tasks.py]: Same here.

File: satnogsclient/scheduler/tasks.py [hunk 15]
```diff
@@ -234,84 +253,99 @@ def ecss_listener(port, queue):
...
             if dict['mode'] == 'cmd_ctrl':
                 print 'Starting ecss feeder thread...'
-                status.MODE = 'cmd_ctrl'
+                os.environ['MODE'] = 'cmd_ctrl'
                 kill_netw_proc()
                 ef = Process(target=ecss_feeder, args=(settings.ECSS_FEEDER_UDP_PORT, settings.ECSS_LISTENER_UDP_PORT,))
                 ef.start()
-                status.ECSS_FEEDER_PID = ef.pid
+                os.environ['ECSS_FEEDER_PID'] = str(ef.pid)
+                print 'Started ecss_feeder process ', ef.pid
```
Comment (johngian) [inline satnogsclient/scheduler/tasks.py]: Same here.

File: satnogsclient/scheduler/tasks.py [hunk 16]
```diff
@@ -234,84 +253,99 @@ def ecss_listener(port, queue):
...
-        if 'switch_mode' in dict.keys() and dict['switch_mode']:
-            print 'Changing mode'
-            if dict['mode'] == status.MODE:
+                os.environ['BACKEND_RX_PID'] = str(rx.pid)
+        if 'mode' in dict.keys():
+            if dict['mode'] == os.environ['MODE']:
                 continue
+            print 'Changing mode'
             if dict['mode'] == 'cmd_ctrl':
                 print 'Starting ecss feeder thread...'
```
Comment (johngian) [inline satnogsclient/scheduler/tasks.py]: Same here.

File: satnogsclient/scheduler/tasks.py [hunk 17]
```diff
@@ -234,84 +253,99 @@ def ecss_listener(port, queue):
...
+                if int(os.environ['ECSS_FEEDER_PID']) != 0:
+                    os.kill(int(os.environ['ECSS_FEEDER_PID']), signal.SIGTERM)
+                    os.environ['ECSS_FEEDER_PID'] = '0'
+                os.environ['SCHEDULER'] = 'ON'
+                scheduler.start()
                 tf = Process(target=task_feeder, args=(settings.TASK_FEEDER_TCP_PORT, settings.TASK_LISTENER_TCP_PORT,))
                 tf.start()
-                status.TASK_FEEDER_PID = tf.pid
+                os.environ['TASK_FEEDER_PID'] = str(tf.pid)
+                print 'Started task feeder process ', tf.pid
```
Comment (johngian) [inline satnogsclient/scheduler/tasks.py]: Same here.

File: satnogsclient/scheduler/tasks.py [hunk 18]
```diff
@@ -234,84 +253,99 @@ def ecss_listener(port, queue):
...
-    if status.TASK_FEEDER_PID != 0:
-        os.kill(status.TASK_FEEDER_PID, signal.SIGTERM)
-        status.TASK_FEEDER_PID = 0
-    scheduler.remove_all_jobs()
+    if int(os.environ['TASK_FEEDER_PID']) != 0:
+        print 'Killing feeder ', int(os.environ['TASK_FEEDER_PID'])
+        os.kill(int(os.environ['TASK_FEEDER_PID']), signal.SIGTERM)
+        os.environ['TASK_FEEDER_PID'] = '0'
+    scheduler.shutdown()
+    print 'Scheduler shutting down!!!!!!!'
```
Comment (johngian) [inline satnogsclient/scheduler/tasks.py]: Same here.

**Comment by johngian** [comment]
Regarding debugging with logging. It's really useful to choose the correct logging level (when using python logging). This way in the future you can choose which messages to see according to DEBUG level. (eg warnings and errors should be captured always, debug and info only when trying to debug something).

Please avoid `print`.

File: satnogsclient/web/app.py [hunk 19]
```diff
@@ -52,13 +53,17 @@ def get_status_info():
 
 @app.route('/control_rx', methods=['GET', 'POST'])
 def get_control_rx():
+    if int(os.environ['ECSS_FEEDER_PID']) == 0:
+        tmp = {}
+        tmp['log_message'] = 'This is a test'
```
Comment (johngian) [inline satnogsclient/web/app.py]: Do we need this here? It's temp code as far as i can understand.

File: satnogsclient/web/app.py [hunk 20]
```diff
@@ -52,13 +53,17 @@ def get_status_info():
...
+        return jsonify(tmp)
     sock = Udpsocket(('127.0.0.1', client_settings.CLIENT_LISTENER_UDP_PORT))
     packet_list = ""
     try:
         conn = sock.send_listen("Requesting received packets", ('127.0.0.1', client_settings.ECSS_FEEDER_UDP_PORT))
+        data = conn[0]
     except:
```
Comment (johngian) [inline satnogsclient/web/app.py]: Same here. Please be more specific on `try/except`.

**Comment by johngian** [comment]
Please do more atomic commits. Also try to keep commit messages as descriptive as possible (always according to some basic policies regarding git message format).

**Pull request merged**